src/pages/charts/data_viewer.js

```diff
@@ -1,101 +1,47 @@
+};
 
 function DataViewer(props) {
-  const { columns, scroll, onDataChange } = props;
-  const tableWidth = scroll.x;
+  const { columns, dataSource, tableWidth, tableHeight, onDataChange, tabIndex } = props;
   const [isEditable, setIsEditable] = useState(false);
-  const mergedColumns = columns.map((column) => {
-    if (column.width) {
-      return column;
-    }
 
+  const mergedColumns = columns.map((column) => {
     return {
       ...column,
+      editable: isEditable,
+      resizable: true,
       width: Math.max(Math.floor((tableWidth - 4) / columns.length), 150),
     };
   });
 
-  const gridRef = useRef();
-
-  const [connectObject] = useState(() => {
-    const obj = {};
-    Object.defineProperty(obj, 'scrollLeft', {
-      get: () => null,
-      set: (scrollLeft) => {
-        if (gridRef.current) {
-          gridRef.current.scrollTo({
-            scrollLeft,
-          });
-        }
-      },
-    });
-    return obj;
-  });
-
-  const renderVirtualList = (rawData, { scrollbarSize, ref, onScroll }) => {
-    ref.current = connectObject;
-    return (
-      <Grid
-        ref={gridRef}
-        className="virtual-grid"
-        columnCount={mergedColumns.length}
-        columnWidth={(index) => {
-          const { width } = mergedColumns[index];
-          return index === mergedColumns.length - 1 ? width - scrollbarSize - 1 : width;
-        }}
-        height={scroll.y}
-        rowCount={rawData.length}
-        rowHeight={() => 54}
-        width={tableWidth}
-        onScroll={({ scrollLeft }) => {
-          onScroll({ scrollLeft });
-        }}
-      >
-        {({ columnIndex, rowIndex, style }) => {
-          const value = rawData[rowIndex][mergedColumns[columnIndex].dataIndex];
-          let InputComponent = Input;
-          let onChange = (event) =>
-            onDataChange(rowIndex, mergedColumns[columnIndex].dataIndex, event.target.value);
-          if (typeof value === 'number') {
-            InputComponent = InputNumber;
-            onChange = (value) =>
-              onDataChange(rowIndex, mergedColumns[columnIndex].dataIndex, value);
-          }
-          return (
-            <div
-              className={classNames('virtual-table-cell', {
-                'virtual-table-cell-last': columnIndex === mergedColumns.length - 1,
-              })}
-              style={style}
-            >
-              {isEditable ? <InputComponent value={value} onChange={onChange} /> : value}
-            </div>
-          );
-        }}
-      </Grid>
-    );
-  };
-
   return (
     <>
       {!isEditable ? (
         <Button onClick={() => setIsEditable(true)}>Edit Data</Button>
       ) : (
         <Button onClick={() => setIsEditable(false)}>Done</Button>
       )}
-      <Table
-        {...props}
-        className="virtual-table"
+
+      <ReactDataGrid
         columns={mergedColumns}
-        pagination={false}
-        showHeader={true}
-        components={{
-          body: renderVirtualList,
```

Comment:
Yes, react data grid provides virtualized rendering and is efficient for huge datasets.
Refer this https://adazzle.github.io/react-data-grid/docs/implementation-notes

Previous implementation has issues when we want to edit data. It is because it re-renders on every change in input component. Edit is not a functionality provided by previous virtual grid and we have to write our own which is failing due to re-renders. Whereas, react-data-grid has inbuilt edit option which, obviously, will not have such issues side it is inbuilt. It also has lot of other features which could be of a great use for us in future.

Reply:
Oh okay